web-console/src/components/segment-timeline/segment-timeline.tsx

```diff
@@ -394,14 +394,16 @@ ORDER BY "start" DESC`;
   onTimeSpanChange = (e: any) => {
     const dStart = new Date();
     const dEnd = new Date();
-    dStart.setMonth(dStart.getMonth() - e);
+    const capabilities = this.props.capabilities;
+    const timeSpan = parseInt(e, 10) || 3;
```

Comment:
Please add some test cases to `web-console/src/components/segment-timeline/segment-timeline.spec.tsx`:
- Default `timeSpan` value of 3
- Custom `timeSpan` value (that is different from default)

Reply:
Something like this is what I had in mind: https://github.com/apache/druid/compare/master...ccaominh:fix-console-segment-timeline-query-timespan. If your fix is reverted so that `this.dataQueryManager.rerunLastQuery()` is called instead of `this.dataQueryManager.runQuery()`, then the test that selects a new period will fail.

Reply:
Thank you for putting this together, this makes sense.